### DISCUSSION TITLE
Some GLTF nodes from `useGLTF` not responding to click events

### DISCUSSION THREAD

**Discussion by neilgamb** [pmndrs/drei#489] [Q&A] [answered]
I am loading a GLTF that I purchased online using the `useGLTF` helper hook and https://gltf.pmnd.rs/ to help generate the appropriate JSX.  As you can see in the `nodes` object, the model includes separate geometries for each tooth as well for each jaw or gums (upper and lower).

I am attempting to register click events for each tooth but for some reason the event is only working for the upper / lower jaw node.  

Please check out my code sample [here](https://codesandbox.io/s/gltfjsx-forked-dhxml?file=/src/Model.js).  

Many thanks in advance for any help! Cheers

**Comment by drcmda** [comment]
i've never seen something like that. it seems to me the gltf is corrupt. i can't open it in blender or gltf.report. https://gltf-viewer.donmccurdy.com opens it but displays 136 errors with missing accessor nodes. if its wrongly indexed or something like that events probably don't come through. 

drop it in here, that's the official spec validator and the model is broken, that it displays at all is a wonder https://github.khronos.org/glTF-Validator

**Reply by neilgamb** [reply]
Great — thank for you the quick reply.  Yes definitely seems like that is the case, thanks for referencing these tools.  Fairly new to using GLTFs and 3D in general. 

Must be the result of my attempt to take the original model and break / separate into subcomponents.  I will look into validating my model.

Thanks again

**Comment by neilgamb** [comment, marked as answer]
Turns out the problem was that I was exporting the model from blender as a `.gltf` and not a `.glb`... original model was a glb

**Reply by drcmda** [reply]
a small hint, you dont need the setState to get the nodes, just map over 

```jsx
{Object.values(nodes).map(node => <mesh key={node.uuid} geometry={node.geometry} ... />)}
```

and r3f can have event handlers in groups. you could have one event handler in the parent group instead of 30 on each object. you will still get the actual mesh that was hit from the event data (e.object). if you forward name={node.name} you'll also be able to identify them.
